test/test_torch.py

```diff
@@ -4147,6 +4147,70 @@ def test_serialization_offset(self):
         self.assertEqual(i, i_loaded)
         self.assertEqual(j, j_loaded)
 
+    def test_serialization_dill_version(self):
+        try:
+            import dill
```

Comment:
This is nicer as a `@skipIf` condition -- instead of the test returning success when dill is not available, it should say it has been skipped. This way if our CI doesn't have dill installed and someone notices we can fix it.

We can do this by making a global variable `TEST_DILL`

Reply:
Also, is dill installed in our CI right now?

Reply:
Looks like it isn't. Could be added to the main `conda install` command in `.circleci/config.yml` (like `requests`) or to `docker/common/install_travis_python.sh` which is where some optional test-time dependencies go (e.g. `hypothesis`, `pillow`).

Do you have any preference for one or the other @zou3519?

Reply:
`dill` is a test-time dependency so I have a preference to put it with the other test-time dependencies in `docker/common/install_travis_python.sh`. However, I think that adding it into that file might require us to rebuild our docker images, which might take a while. Let me look into this a little more...

Reply:
I'm going to call for help here. cc @yf225 do we have to rebuild docker images if we add another package (dill) to https://github.com/pytorch/pytorch/blob/master/.circleci/docker/common/install_travis_python.sh? 

I think the answer is yes because the only usages of that script are in the Dockerfile: https://github.com/pytorch/pytorch/blob/master/.circleci/docker/ubuntu/Dockerfile#L50

Reply:
@zou3519 Yes I think we have to rebuild the docker image because of the reason you mentioned :)

Reply:
Regarding this thread, @kurtamohler I need to rebuild the docker images for something else anyways, so I'll add the dill dependency in a separate PR.

I'll be adding `dill>=0.3.1` (or maybe it should be `dill==0.3.1` ?), does that work?

Reply:
@zou3519, yeah, sounds good to me, thanks! It might be better to do `dill>=0.3.1`. That way we'll  be able to find out more quickly if a future version of dill has breaking changes.